relay-server/src/services/store.rs

```diff
@@ -363,19 +371,26 @@ impl StoreService {
             return Ok(());
         }
 
-        let remote_addr = envelope.meta().client_addr().map(|addr| addr.to_string());
+        // FIXME: This error propagation happens in almost all the cases,
+        // except when using `TrackOutcome` nested deep in some of the called functions.
+        // Might be worth doing this propagation earlier.
+        let event_id = event_id.ok_or(StoreError::NoEventId)?;
+        let project_id = scoping.project_id;
 
-        let kafka_messages = Self::extract_kafka_messages_for_event(
-            event_item.as_ref(),
-            event_id.ok_or(StoreError::NoEventId)?,
-            scoping,
-            start_time,
-            remote_addr,
-            attachments,
-        );
+        if let Some(event_item) = event_item {
```

Comment:
There is a check right above:

```rs
        if event_item.is_none() && attachments.is_empty() {
            // No event-related content. All done.
            return Ok(());
        }
```

Which signals to me it's possible to have no `event_item` but you have attachements, with this new code the attachments are not handled anymore (?).

Reply:
maybe its a similar confusion that @loewenheim mentioned:
producing *individual* attachments happens in `produce_attachment_chunks` now.
otherwise the logic is the same as before, but I can also clean that up further as you are right that the above check is redundant now.